### PR TITLE
Attendance model

### PR DESCRIPTION
**Pull request by robertvictoreen**

### CODE REVIEW

File: src/main/java/seedu/address/model/attendance/AttendanceMark.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    public static final String MESSAGE_CONSTRAINTS =
+            "AttendanceMark should be a 0 or a 1";
+
+    /**
+     * Constructs a {@code AttendanceMark}.
+     *
+     * @param mark A valid real number.
+     */
+    public AttendanceMark(String mark) {
+        super(mark);
```
Comment (spencertan96) [inline src/main/java/seedu/address/model/attendance/AttendanceMark.java]: Ehh you don't check if mark given was 1 or 0 here? Why not isValid here?

File: src/main/java/seedu/address/model/attendance/exceptions/DuplicateAttendanceException.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package seedu.address.model.attendance.exceptions;
+
+/**
+ * Signals that the operation will result in duplicate Attendance
+ * (Attendance are considered duplicates if they have the same
+ * identity).
+ */
+public class DuplicateAttendanceException extends RuntimeException {
+    public DuplicateAttendanceException() {
+        super("Operation would result in duplicate attendance sessions");
```
Comment (spencertan96) [inline src/main/java/seedu/address/model/attendance/exceptions/DuplicateAttendanceException.java]: full stop? :x

File: src/test/java/systemtests/AddCommandSystemTest.java [hunk 3]
```diff
@@ -136,22 +135,6 @@ public void add() {
         command = PersonUtil.getAddCommand(HOON) + " " + PREFIX_TAG.getPrefix() + "friends";
         assertCommandFailure(command, AddCommand.MESSAGE_DUPLICATE_PERSON);
 
-        /* Case: missing name -> rejected */
```
Comment (spencertan96) [inline src/test/java/systemtests/AddCommandSystemTest.java]: Can leave missing name in I think, names are supposed to be compulsory

**Comment by spencertan96** [comment]
Is the AddCommandParserTest fixed? I think it was that test that failed

**Comment by robertvictoreen** [comment]
> Is the AddCommandParserTest fixed? I think it was that test that failed

@spencertan96 Yes it is fixed now, it was some of the tests and also the Index in the AddCommand caused some tests to fail because the lastShownList may be different when the "find" command is used.

File: src/main/java/seedu/address/model/attendance/exceptions/DuplicateAttendanceException.java (hunk 2 above)
Comment (robertvictoreen) [reply src/main/java/seedu/address/model/attendance/exceptions/DuplicateAttendanceException.java]: Sure thanks, although it is copied from the DuplicatePersonException

File: src/main/java/seedu/address/model/attendance/AttendanceMark.java (hunk 1 above)
Comment (robertvictoreen) [reply src/main/java/seedu/address/model/attendance/AttendanceMark.java]: I followed the implementation of Mark, and the isValid is tested in ParserUtil

File: src/test/java/systemtests/AddCommandSystemTest.java (hunk 3 above)
Comment (robertvictoreen) [reply src/test/java/systemtests/AddCommandSystemTest.java]: Yes thanks!

File: src/main/java/seedu/address/model/attendance/AttendanceMark.java (hunk 1 above)
Comment (spencertan96) [reply src/main/java/seedu/address/model/attendance/AttendanceMark.java]: Okay!

**Pull request merged**